### ISSUE TITLE
Items that would be destroyed by explosion traps won't behave like this if they fall onto the traps from the previous floor.

### ISSUE THREAD

**Issue by koisikawa** [00-Evan/shattered-pixel-dungeon#1887]

**Comment by 00-Evan** [comment]
Which item didn't behave in the way you expected it? The game uses standard item dropping logic and so there shouldn't be any difference. It's worth noting that bombs and explosion effects were changed in v2.5 though, and explosions no longer destroy any equipment items.

**Comment by koisikawa** [comment]
The "items" I refered in the title are the items that can still be destroyed by explosion in v2.5, such as scrolls.

And, based on what I have observed, I am certain that such an issue exists. If you have doubts about it, I can provide a detailed description.

**Comment by 00-Evan** [comment]
I'm afraid I'm not able to reproduce this. From some testing falling consumables such as scrolls are correctly destroyed by explosive traps. The only exceptions to this are items like scrolls of upgrade, which have always been immune to destruction.

**Comment by koisikawa** [comment]
🤔I tried to reproduce it today by a modified version (just to debug easily, without modifying the game logic), but it was successfully reproduced. 

But since it's a minor issue that is difficult to encounter in a normal game and does little harm for the game, just stop here.

**Comment by 00-Evan** [comment]
Specifically the bug you're seeing is:
- item which would normally be destroyed by an explosion is thrown into chasm
- on the next floor, item falls onto explosive trap by chance
- trap is triggered, explosion happens, item is unaffected.

That's what I tested (with a small modification to guarantee that an explosive trap is where the item lands) and the item was destroyed every time. Are you sure your modification does not affect game logic, or cause events to occur in a way that's impossible in-game? I often get 'bug reports' from people using modifications to Shattered that are not possible in the game itself.

**Comment by koisikawa** [comment]
I want to put aside the question about the modified version for now because of the latest discovery I made: **This is likely a misunderstanding caused by an extremely rare coincidence.**

The content of the video where the "bug" occurs, namely the basis for the issue I raise, is: When the player went downstairs, he heard an explosion and speculated that it was triggered by an item he thrown from upstairs. But when he walked to the room with the explosion traps, he found that the scroll (scroll of challenge) he had thrown from upstairs was safely on a triggered trap, as if the scroll had not been destroyed by the explosion.

I watched the video again and recorded every thrown item upstairs, comparing it with the items collected downstairs, and found that a shattered honeypot which was thrown first was missing. Based on this, it is highly likely that the honeypot and the scroll coincidentally fell into the same place, and coincidentally both fell into the explosion trap, causing the honeypot thrown first to be destroyed by the explosion and the scroll thrown later to remain safe.

As for my reproducing experiment, I just generated a lot of food by debugging and threw them one by one into the downstairs with explosion traps, and found that there were indeed "undestroyed" food on the triggered traps downstairs. But in fact it was just a coincidence that was the same as what I had observed before.

Sorry, my fault.🙁
